backend/subroker/views.py

```diff
@@ -1,3 +1,392 @@
 from django.shortcuts import render
+from django.db.models.query import EmptyQuerySet
+from django.http import HttpResponse, HttpResponseNotAllowed, HttpResponseBadRequest, JsonResponse
+from django.contrib.auth.models import User
+from django.contrib.auth import login as auth_login, logout
+from django.views.decorators import csrf
+from django.views.decorators.csrf import ensure_csrf_cookie
+import json
+from json.decoder import JSONDecodeError
+from .models import Ott, Group, Content, Review
+from django.core.exceptions import ObjectDoesNotExist
+
+
+#signup/ : User Sign Up
+def signup(request):
+    if request.method == 'POST':
+        req_data = json.loads(request.body.decode())
+        username = req_data['username']
+        password = req_data['password']
+        User.objects.create_user(username, password)
+        return HttpResponse(status=201)
+    else:
+        return HttpResponseNotAllowed(['POST'])
+
+#token/ : Token
+@ensure_csrf_cookie
+def token(request):
+    if request.method == 'GET':
+        return HttpResponse(status=204)
+    else:
+        return HttpResponseNotAllowed(['GET'])
+
+#login/ : User Log In
+def login(request):
+    if request.method=='POST' :
+        req_data = json.loads(request.body.decode())
+        username = req_data['username']
+        password = req_data['password']
+        try:
+            user = User.objects.get(username = username)
+        except (User.DoesNotExist) as e:
+            return HttpResponse(status=401)
+        if user.check_password(password) == False:
+            return HttpResponse(status=401)
+        auth_login(request, user)
+        return HttpResponse(status=204)
+
+    else:
+        return HttpResponseNotAllowed(['POST'])
+
+#logout/ : User Log Out
+def logout(request):
+    if request.method == 'GET':
+        if request.user.is_authenticated:
+            logout(request)
+            return HttpResponse(status=204)
+        else:
+            return HttpResponse(status=401)
+    else:
+        return HttpResponseNotAllowed(['GET'])
+
+#Group
+def group_list(request):    
+    if request.method == 'GET':
+        if request.user.is_authenticated:
+            group_all_list = [group for group in Group.objects.all().values()]
+            if not group_all_list:
+                return HttpResponse(status=404)
+            return JsonResponse(group_all_list, safe=False, status=200)
+        else:
+            return HttpResponse(status=401)
+        
+    elif request.method == 'POST':
+        if request.user.is_authenticated:
+            try:
+                req_data = json.loads(request.body.decode())
+                group_name = req_data['name']
+                group_description = req_data['description']
+                group_is_public = bool(req_data['is_public'])
+                group_password = int(req_data['password'])
```

Comment:
password is optional, so jsonDecodeError should skip password, or some special integer should be put here(like -1)